base/system/smss/sminit.c

```diff
@@ -1819,9 +1819,9 @@ SmpCreateDynamicEnvironmentVariables(VOID)
     if (NT_SUCCESS(Status))
     {
         /* To combine it into a single string */
-        swprintf((PWCHAR)PartialInfo->Data + wcslen((PWCHAR)PartialInfo->Data),
-                 L", %S",
-                 PartialInfo2->Data);
+        ((PWCHAR)(PartialInfo->Data))[PartialInfo->DataLength / sizeof(WCHAR)] = L'\0';        
+        wcsncat((PWCHAR)PartialInfo->Data, L", ", 2 * sizeof(WCHAR));
```

Comment:
Simply use `wcscpy()`.

Reply:
How is it superior to wcsncat when the intended action is combining strings ?
wcsncat based implementation works perfectly, while previous design based on swprintf fails.
What would be the benefit from doing this just in a different manner ?

Reply:
These 2 lines are overly complicated/confusing:
writing a `UNICODE_NULL` is useless as you overwrite it immediately after,
using `wcsncat` for a fixed length string is overkill,
moreover it expects a count of characters, not bytes,
...

As you were told, `swprintf` itself was just fine, the issue was `wcslen` use.

Reply:
"overly complicated/confusing" : Looks more like a self-esteem problem given the simplicity of the proposed design :-)

All jokes appart, i'll rework again with swprintf if this is a "must" to get it merged into master.

Reply:
> jokes

Maybe that is part of the problem: it seems you rewrote these lines multiple times, yet new code is (still) incorrect! And you don't seem to get the point :-(

Reply:
If you were thinking about 
swprintf((PWCHAR)PartialInfo->Data + PartialInfo->DataLength/sizeof(WCHAR)-1,
           		L", %*S",
         		PartialInfo2->DataLength/sizeof(WCHAR),
			PartialInfo2->Data);

It does not work %*S just adds "blanks" before PartialInfo2->Data first character and %S just prints the 1st character, as current broken design.

Since the proposed design if fit-for-purpose, simple, safe and with an adequate level of performance with regard to the need, I suggest to go with it.

Should a dev prefer to overwrite and commit a proposal of it's own based on swprintf or any different design, with a proper justification, i'm perfectly fine.

So @SergeGautherie, @HBelusca either
- You approve my code and merge it into master 
- You commit a different code and merge it into master 
- You place this PR on hold and keep the current broken code, which is in any way worse than any above options.